### DISCUSSION TITLE
unsafeHTML does not correctly render event attributes like @click

### DISCUSSION THREAD

**Discussion by wodecinghaolen** [lit/lit#4661] [Q&A] [answered]
If I were to pass in something like the following,
<calculate
  style="background-color:white"
  @check=${this._handleCheck}
></calculate>
Where this._handleCheck is a method of the component that encapsulate this calculate component.

It would take the method and simply replace this._handleCheck. The following is what I get in the browser. The _handleCheckfunction is not called when the check event is fired.
<calculator 
style="background-color:white" 
@check="_handleCheck(e)" {="" console.log("event="" check="" fired!");="" }=""></calculator>

Is it intended or potentially a bug? How do I correctly render the calculator component with unsafeHTML? Any input would be much appreciated!

**Comment by augustjk** [comment, marked as answer]
This is intended. `unsafeHTML()` is meant to include just HTML strings. Things like `@click` are `lit-html` template bindings which are not purely HTML things.
